ci-operator/templates/openshift/installer/cluster-launch-installer-e2e.yaml

```diff
@@ -153,10 +153,10 @@ objects:
 
         i=0
         MAX_RETRIES=10
-        until oc wait "${ROUTER_DEPLOYMENT}" -n "${ROUTER_NAMESPACE}" --for condition=available --timeout=10m || [ $i -eq $MAX_RETRIES ]; do
+        until oc --request-timeout=2m rollout status "${ROUTER_DEPLOYMENT}" -n "${ROUTER_NAMESPACE}" -w || [ $i -eq $MAX_RETRIES ]; do
             i=$((i+1))
-            [ $i -eq $MAX_RETRIES ] && echo "timeout waiting for router to be available" && exit 1
-            echo "error deploy/router did not come up"
+            [ $i -eq $MAX_RETRIES ] && echo "timeout waiting for ${ROUTER_NAMESPACE}/${ROUTER_DEPLOYMENT} to be available" && exit 1
+            echo "error ${ROUTER_NAMESPACE}/${ROUTER_DEPLOYMENT} did not come up"
             sleep 60
```

Comment:
60 seconds seems like a long break for a [network or spurious issue][1].  On the other hand, if we only `sleep 1`, we'd `exit 1` here if the network was out for only 10 seconds.  So I'd still rather have logic that said "this is the wall time we're willing to wait, we don't care how many network connections you need to cover that window".  If you don't like [my earlier proposal][2], how about:

```sh
TARGET="$(date -d '10 minutes' +%s)"
NOW="$(date +%s)"
while [[ "${NOW}" -lt "${TARGET}" ]]; do
  REMAINING="$((TARGET - NOW))"
  if oc --request-timeout="${REMAINING}s" oc rollout status "${ROUTER_DEPLOYMENT}" -n "${ROUTER_NAMESPACE}" -w; then
    break
  fi
  NOW="$(date +%s)"
done
[[ "${NOW}" -ge "${TARGET}" ]] && echo "timeout waiting for ${ROUTER_NAMESPACE}/${ROUTER_DEPLOYMENT} to be available" && exit 1
```

or:

```sh
TARGET="$(date -d '10 minutes' +%s)"
NOW="$(date +%s)"
REMAINING="$((TARGET - NOW))"
until oc --request-timeout="${REMAINING}s" oc rollout status "${ROUTER_DEPLOYMENT}" -n "${ROUTER_NAMESPACE}" -w; do
  NOW="$(date +%s)"
  REMAINING="$((TARGET - NOW))"
  if [[ "${REMAINING}" -le 0 ]]; then
    echo "timeout waiting for ${ROUTER_NAMESPACE}/${ROUTER_DEPLOYMENT} to be available"
    exit 1
  fi
done
```

And all of these are untested sketches, if we hit on phrasing you like let me know and I can test it more thoroughly ;).

[1]: https://github.com/openshift/release/pull/2004#discussion_r227951274
[2]: https://github.com/openshift/release/pull/2004#discussion_r227924803

Reply:
Nah, the original one is fine. I'll use that one (and test it) and update this  PR. Thanks.